docs/DevGuide/BuildSystem.md

```diff
@@ -69,15 +69,27 @@ add_spectre_library(${LIBRARY} ${LIBRARY_SOURCES})
 
 target_link_libraries(
   ${LIBRARY}
-  <list_of_interfaces>
+  PUBLIC
+  <list_of_public_libraries>
+  PRIVATE
+  <list_of_private_libraries>
+  INTERFACE
+  <list_of_interface_libraries>
   )
 ```
-where `<list_of_interfaces>` is an alphabetized list of lines of the form
+where each `<list_of_X_libraries>` is an alphabetized list
+of libraries of the form
 ```
-  INTERFACE SOME_LIBRARY
+  SOME_LIBRARY
+  SOME_OTHER_LIBRARY
+  YET_ANOTHER_LIBRARY
 ```
-where `SOME_LIBRARY` is a library that must be linked in order for the
-new library to be successfully linked in an executable.
+The libraries listed under `INTERFACE` are those included in at
+least one `.hpp` file in `LIBRARY` but never used in any `.cpp` files
+in `LIBRARY`.  The libraries listed under `PRIVATE` are used in
+at least one `.cpp` file in `LIBRARY` but not in any `.hpp` file
+in `LIBRARY`.  The libraries listed under `PUBLIC` are used in at
+least one `.hpp` file and at least one `.cpp` file in `LIBRARY`.
```

Comment:
I suggest the following additional sentence:

> Note that a library also counts as being used in a `.cpp` file if the corresponding `.hpp` file includes it.

You could also add something like this:

> In other words, list a dependency as `PRIVATE` if it is only needed to compile the library. List a dependency as `INTERFACE` if it is _not_ needed to compile the library, but only for including headers. List a dependency as `PUBLIC` if it is needed for both.